docker/README.md

```diff
@@ -2,32 +2,65 @@
 
 ## Prerequisites
 
+These instructions were tested on Ubuntu Jammy (22.04). Other recent versions
+should work as well.
+
 If your machine has an NVIDIA GPU, install Docker and NVIDIA Docker 2 following
 instructions [here](https://github.com/osrf/subt/wiki/Docker%20Install).
 
 If you don't have an NVIDIA GPU, Gazebo GUI will use software rendering via
-Mesa llvmpipe (you can check that in ~/.gz/rendering/ogre2.log after starting
+Mesa llvmpipe (you can check that in `~/.gz/rendering/ogre2.log` after starting
 the Gazebo GUI).
 
-## Example usage
+## Pull from DockerHub
+
+This image (and possibly some extra data) is built and pushed to DockerHub under
+[osrf/icra2023_ros2_gz_tutorial](https://hub.docker.com/r/osrf/icra2023_ros2_gz_tutorial/tags).
+
+You can pull it from DockerHub and skip building it locally:
+```
+docker pull osrf/icra2023_ros2_gz_tutorial:<tag>
+```
+Replace `<tag>` with `tutorial_nvidia` or `tutorial_no_nvidia`, depending on whether you
+have an NVIDIA GPU.
+
+## Build the image locally
 
-To build an image:
+Skip this step if you already pulled the image from DockerHub.
 
 If you have an NVIDIA graphics card, build using the NVIDIA Docker base image:
 ```
 ./build.bash nvidia_opengl_ubuntu22
 ./build.bash icra2023_tutorial
 ```
+
 Otherwise, build without NVIDIA Docker:
 ```
 ./build.bash icra2023_tutorial --no-nvidia
 ```
 
-To spin up a container from an image:
+A few tags will be created for the same image, for convenience of running
+commands.
+
+## Run the image
+
+To spin up a container from an image, this should work whether you pulled from
+DockerHub or built locally:
+```
+# If you have NVIDIA GPU
+./run.bash osrf/icra2023_ros2_gz_tutorial:tutorial_nvidia
+
+# Without NVIDIA GPU
+./run.bash osrf/icra2023_ros2_gz_tutorial:tutorial_no_nvidia --no-nvidia
+```
+
+If you built the image locally, this is a shorter, equivalent command:
 ```
 ./run.bash icra2023_tutorial
 ```
```

Comment:
I'd honestly get rid of this.  The longer commands are fine (people are just going to cut-n-paste), and I think this can only lead to confusion.  But I'll leave it up to you.

Reply:
Yeah... I thought about just having one. But this command is very convenient for someone who knows what's going on, because you can tab-complete the directory name and don't need to copy-paste. That's how I usually do it. Maybe I should add that.

The `osrf/` one is only there because that's the name of the image on DockedHub, and people who pull the image will need to use that. It's very long, and copy-pasting means people need to run `docker images` every time to copy it. Maybe we should add a line setting up an alias for quick access.

Reply:
Added comments in in 97b343b